src/v/cluster/rm_stm.cc

```diff
@@ -262,39 +271,85 @@ ss::future<checked<model::term_id, tx::errc>> rm_stm::do_begin_tx(
           synced_term, producer, std::nullopt, _sync_timeout);
         if (ar != tx::errc::none) {
             vlog(
-              _ctx_log.trace,
-              "can't begin tx {} because abort of a prev tx {} failed with {}",
+              _ctx_log.warn,
+              "[{}] can't begin tx because abort of a prev tx with lower epoch "
+              "{} failed with {}",
+              *producer,
               pid,
-              current_pid,
               ar);
             co_return tx::errc::stale;
         }
     } else if (producer->has_transaction_in_progress()) {
-        // check for duplicate request
-        if (
-          producer->transaction_state()->status
-          == tx::partition_transaction_status::initialized) {
+        const auto& tx_state = producer->transaction_state();
+        if (tx_seq < tx_state->sequence) {
+            vlog(
+              _ctx_log.warn,
+              "[{}] Fencing begin_tx request due to a smaller sequence "
+              "number ",
```

Comment:
nit: let's be more explicit which sequence number it is, it may be confusing as there are two sequence numbers tracked in the `rm_stm`